### DISCUSSION TITLE
How to call VAPIX API with Digest Authentication in Javascript

### DISCUSSION THREAD

**Discussion by vivekatoffice** [AxisCommunications/discussions#73] [VAPIX] [answered]
# How to call VAPIX API with Digest Authentication in JavaScript(JS).

## I am trying to do something like below but not succeeded:

```JS
var xhr = new XMLHttpRequest();
xhr.withCredentials = true;

xhr.addEventListener("readystatechange", function() {
  if(this.readyState === 4) {
    console.log(this.responseText);
  }
});

xhr.open("GET", "http://192.168.50.14/axis-cgi/param.cgi?action=list&group=properties.audioclip.playoptions");

xhr.send();  

```

# In my project I am only using HTML and JS

**Comment by mattias-kindborg-at-work** [comment, marked as answer]
Hi @axisvivek!

I don't think `XMLHttpRequest` has native support for [digest access authentication](https://en.wikipedia.org/wiki/Digest_access_authentication). A quick search tells me that this easily is done as long as you implement the protocol of digest authentication. My guess would be that you extract `realm`, `nounce` and `qop` from the Axis device challenge response and build your `Authorization` header based on that.
